contracts/base/strategies/beethoven/BeethovenFDBase.sol

```diff
@@ -0,0 +1,132 @@
+*/
+pragma solidity 0.8.4;
+
+import "../../../third_party/beethoven/IFBeets.sol";
+import "../../../base/strategies/beethoven/BeethovenBase.sol";
+
+/**
+  @author OlegN
+  @title A strategy for The Fidelio Duetto beethoven pool
+  @dev This strategy is using extra step (fBeets) to obtain extra rewards.
+  It requires to add extra exchange rate logic
+*/
+contract BeethovenFDBase is BeethovenBase {
+  using SafeMath for uint;
+  using SafeERC20 for IERC20;
+
+  uint private constant _PRECISION = 10 ** 18;
+
+  /// @notice token to get extra rewards by wrapping BPT tokens
+  address fBeetsToken;
+
+  constructor(
+    address _controller,
+    address _underlying,
+    address _vault,
+    address[] memory __rewardTokens,
+    address _pool,
+    uint _poolId,
+    address _beethovenVault,
+    address _depositToken,
+    bytes32 _beethovenPoolId,
+    bytes32 _rewardToDepositPoolId,
+    address _fBeetsToken
+
+  ) BeethovenBase(
+    _controller,
+    _vault,
+    _underlying,
+    __rewardTokens,
+    _pool,
+    _poolId,
+    _beethovenVault,
+    _depositToken,
+    _beethovenPoolId,
+    _rewardToDepositPoolId
+  ){
+    fBeetsToken = _fBeetsToken;
+  }
+
+  /// @notice Claim rewards from external project and send them to FeeRewardForwarder
+  /// @dev this method includes try..catch part because of IBeethovenxChef(pool).harvest could fail with the
+  ///      low amount of rewards like 1. We could have such values due to rounding.
+  function doHardWork() external override onlyNotPausedInvesting restricted {
+    investAllUnderlying();
+    try IBeethovenxChef(pool).harvest(poolId, address(this)){} catch {}
+    liquidateReward();
+  }
+
+  /// @notice Strategy balance in the MasterChef pool
+  /// @return bal Balance amount in underlying tokens
+  function rewardPoolBalance() public override view returns (uint) {
+    (uint _fBeetsAmount,) = IBeethovenxChef(pool).userInfo(poolId, address(this));
+    return underlyingForFBeets(_fBeetsAmount);
+  }
+
+  /// @notice TVL of the underlying in the MasterChef pool
+  /// @dev Only for statistic
+  /// @return Pool TVL
+  function poolTotalAmount() external view override returns (uint) {
+    return IERC20(_underlyingToken).balanceOf(pool) + rewardPoolBalance();
+  }
+
+  /// @dev Deposit wrapped underlying (fBeets) to MasterChef pool
+  /// @param amount Deposit amount
+  function depositToPool(uint amount) internal override {
+    IERC20(_underlyingToken).safeApprove(fBeetsToken, 0);
+    IERC20(_underlyingToken).safeApprove(fBeetsToken, amount);
+    IFBeets(fBeetsToken).enter(amount);
+    IERC20(fBeetsToken).safeApprove(pool, 0);
+    IERC20(fBeetsToken).safeApprove(pool, balanceFBits());
+    IBeethovenxChef(pool).deposit(poolId, balanceFBits(), address(this));
+  }
+
+  /// @dev Withdraw fBits from MasterChef pool and unwrap underlying
+  /// @param amount Deposit amount
+  function withdrawAndClaimFromPool(uint amount) internal override {
```

Comment:
override exit function for withdraw 100% available amount

Reply:
Done